sql-plugin/src/main/scala/com/nvidia/spark/rapids/asyncProfiler.scala

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.io.{BufferedInputStream, BufferedOutputStream, FileInputStream, FileOutputStream}
+import java.nio.file.{Files, Paths}
+import java.util.concurrent.{ConcurrentHashMap, Executors, ScheduledExecutorService, ScheduledFuture, TimeUnit}
+import java.util.zip.GZIPOutputStream
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+import com.nvidia.spark.rapids.Arm.withResource
+import one.profiler.{AsyncProfiler, AsyncProfilerLoader}
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.TaskContext
+import org.apache.spark.api.plugin.PluginContext
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.SerializableConfiguration
+
+/**
+ * Unlike ProfilerOnExecutor, which is used to create Nsys reports, this class is used to create
+ * flame graph to understand where time is spent for each stage. Typically, we can use metrics to
+ * achieve the same goal, but metrics cannot guarantee to capture all the time spent in a stage,
+ * and it may not be detailed enough because it lacks the stacktraces information.
+ *
+ * From the perspective of profiling, we know that different stages may exhibit different patterns.
+ * Since the output is a FUSED flame graph, it is not possible to distinguish between different
+ * stages or threads, so the profiling is done per stage, and the output file is named
+ * after each stage.
+ *
+ * It is strongly recommended to set 'spark.scheduler.mode' to 'FIFO' for better stage
```

Comment:
besides the code comments, we can also log some warnings if it's non FIFO case.

Reply:
added a warning log at https://github.com/NVIDIA/spark-rapids/pull/12960/commits/b600a8e580c6e1ccd8e2505c712f1e2dd8ccfeb1